File: Python/_475_Heaters/solution.py
class Solution:
    def findRadius(self, houses, heaters):
        """
        :type houses: List[int]
        :type heaters: List[int]
        :rtype: int
        """
        houses = sorted(houses)
        heaters = sorted(heaters)
        i, radius = 0, 0
        for house in houses:
            while i < len(heaters) - 1 and house > heaters[i]:
                i += 1
            if i > 0:
                radius = max(radius, min(abs(heaters[i] - house), house - heaters[i - 1]))
            else:
                radius = max(radius, abs(house - heaters[0]))
        return radius

File: Java/_475_Heaters/Solution.java
class Solution {
    public int findRadius(int[] houses, int[] heaters) {
        Arrays.sort(houses);
		Arrays.sort(heaters);
		int i = 0;
		int	radius = 0;
        // int i = 0, radius = 0;
		for (int house : houses) {
			while (i < heaters.length - 1 && house > heaters[i])
				i++;
			if (i > 0)
				radius = Math.max(radius, Math.min(Math.abs(heaters[i] - house), house - heaters[i - 1]));
			else
				radius = Math.max(radius, Math.abs(house - heaters[0]));
		}
		return radius;
    }
}